Feat: case widgets

mbritense opened the pull request.

theo-ritense committed:
109484: CW admin add/edit tab (#1345)

Added widget tab support

marijnritense committed:
Feature/111002 case widgets admin tab widgets (#1346)

* Added CaseWidgetTabWidgets

* added pbac

* PR comments fixes

* Fixes, mapper and tests related to widget inheritance.

---------

Co-authored-by: Theo <theo.beers@ritense.com>

theo-ritense committed:
Added case widget user tab (#1353)

* Added case widget user tab

theo-ritense committed:
109495 fields widget (#1357)

* Fields widget

theo-ritense committed:
Added runWithoutAuthorization for retrieving data since PBAC will be applied on a widget level already. (#1359)

marijnritense committed:
added auto deployment for case widget tabs (#1361)

marijnritense committed:
Added export for case widget tabs (#1362)

ThomasMinkeRitense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Fix for the read-only transaction exception via the outbox (#1364)

marijnritense committed:
added import for case widget tabs (#1363)

theo-ritense committed:
109496 case widget table (#1369)

Added case widget: table

Klaas-Ritense committed:
#113976 CW admin: add createdOn & createdBy

theo-ritense committed:
Added autodeployment and fixed null collection error (#1376)

* Added autodeployment and fixed null collection error

Klaas-Ritense committed:
Merge branch 'feature/109355-case-widgets' into story/113976-CW-admin-add-createdOn-createdBy

theo-ritense committed:
Fixed autodeployment file (#1379)

Klaas-Ritense committed:
Merge branch 'feature/109355-case-widgets' into story/113976-CW-admin-add-createdOn-createdBy

Klaas-Ritense committed:
Update case/src/main/resources/config/liquibase/changelog/20240527-add-createdon-createdby-to-case-tabs.xml

Co-authored-by: theo-ritense <95682624+theo-ritense@users.noreply.github.com>

teodor-ritense committed:
Merge pull request #1375 from valtimo-platform/story/113976-CW-admin-add-createdOn-createdBy

#113976 CW admin: add createdOn & createdBy

ivo-ritense committed:
Added widget authorization (#1371)

mbritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

mbritense committed:
feat: add case widget tab permission autodeployment (#1387)

* add permission

* add missing role key

mbritense committed:
add custom case widget option

mbritense committed:
Merge pull request #1392 from valtimo-platform/feature/109875-custom-case-widget-tab

feat: custom case widget

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
109498: FormIo case widget (#1393)

Co-authored-by: valtimo-platform[bot] <80107705+valtimo-platform[bot]@users.noreply.github.com>

theo-ritense committed:
Added firstColumnAsTitle property for table widget (#1396)

ivo-ritense committed:
Added collection widget type (#1408)

theo-ritense committed:
Use reflection to get generic type arguments for CaseWidgetMapper (#1406)

* Added default implementation for CaseWidgetMapper types, using reflection to return the value.

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Story/109503 fixes (#1414)

* Fix for Optional error on casetab to documentdefinition mapping

* - widget security fixes
- added collection widget to autodeployment

* mapper will throw an error when JsonSchemaDocumentDefinition can not be found by name.

theo-ritense committed:
Added an classpath override for the KeycloakClientRequestFactory so that it will not be picked up by component-scanning.

theo-ritense committed:
added width property for collection widget fields (#1418)

theo-ritense committed:
Changed formDefinition to formDefinitionName for formio case widget (#1412)

Co-authored-by: valtimo-platform[bot] <80107705+valtimo-platform[bot]@users.noreply.github.com>

theo-ritense reviewed (dismissed):
Blocking this PR until it's done

Klaas-Ritense committed:
fix/three-fields-issue (#1425)

mbritense committed:
Merge branch 'next-minor' into feature/merge-next-minor-case-widgets

# Conflicts:
#	form/src/main/kotlin/com/ritense/form/autoconfigure/FormAutoConfigurationKotlin.kt

mbritense committed:
refactor

theo-ritense committed:
Merge pull request #1428 from valtimo-platform/feature/merge-next-minor-case-widgets

feat: next-minor into case-widgets

marijnritense committed:
Story/113212 cw authorization (#1427)

authorization for context sensitive checks
---------

Co-authored-by: valtimo-platform[bot] <80107705+valtimo-platform[bot]@users.noreply.github.com>

mbritense committed:
add show_tasks property to case tabs

mbritense committed:
Update case/src/main/kotlin/com/ritense/case/domain/CaseTab.kt

Co-authored-by: Klaas Schuijtemaker <94360980+Klaas-Ritense@users.noreply.github.com>

mbritense committed:
Merge pull request #1431 from valtimo-platform/feature/115539-show-tasks-on-case-tabs

feat: show tasks on case tabs

mbritense committed:
change id

mbritense committed:
Merge pull request #1435 from valtimo-platform/feature/change-permission-changeset-id

feat: change case widget changeset id

theo-ritense committed:
115967: Added title field to collection widget (#1437)

* Added title field to collection widget

theo-ritense committed:
fix for TemporaryResourceStorageDeletionServiceIntegrationTest (#1444)

* DisabledTemporaryResourceStorageDeletionServiceIntegrationTest

theo-ritense committed:
Fields data resolver will not return null values anymore. (#1446)

marijnritense committed:
fixed contexts being ignored when updating permissions via REST

marijnritense committed:
added mapping to include context container and context resource types

theo-ritense committed:
Testing branch protections

theo-ritense committed:
Exclude empty `contextResourceType` and `contextConditions` fields when serializing (#1456)

Klaas-Ritense committed:
Merge remote-tracking branch 'origin/next-minor' into merge-next-minor-into-feature-case-widgets

Klaas-Ritense committed:
fix merge conflict

Klaas-Ritense committed:
Merge pull request #1460 from valtimo-platform/merge-next-minor-into-feature-case-widgets

Merge next-minor into feature/case-widgets

marijnritense committed:
updated toPredicate and all existing specifications to use the context (#1463)

theo-ritense committed:
Fix "Getter methods of lazy classes cannot be final: com.ritense.case_.domain.tab.CaseWidgetTabWidget#getHighContrast" error (#1466)

theo-ritense committed:
Added `datetime` display type for widgets (#1473)

mbritense committed:
fix read only transactions case tabs service (#1476)

marijnritense committed:
removed the showTasks property from the autodeployment DTO as this is a breaking change (#1486)

theo-ritense committed:
Bugfix/116290 widget key duplication (#1488)

Fixed key duplication issue on case widgets. Widgets with the key are now allowed on different tabs.

ivo-ritense committed:
Added check for allowed characters for tab key (#1492)

theo-ritense committed:
Merge branch 'next-minor' into feature/109355-case-widgets

theo-ritense committed:
Story/118957 case widget width limit (#1493)

Added validations on widget DTO and fixed generic validation annotations.

ThomasMinkeRitense committed:
Merge branch 'next-minor' into feature/109355-case-widgets
